tests/contrib/redis/test_redis_asyncio.py

```diff
@@ -0,0 +1,213 @@
+from ddtrace.vendor.wrapt import ObjectProxy
+from tests.utils import override_config
+
+from ..config import REDIS_CONFIG
+
+
+def get_redis_instance(max_connections: int):
+    return redis.asyncio.from_url("redis://127.0.0.1:%s" % REDIS_CONFIG["port"], max_connections=max_connections)
+
+
+@pytest.mark.asyncio
+@pytest.fixture
+def redis_client():
+    r = get_redis_instance(max_connections=10)  # default values
+    yield r
+
+
+@pytest.mark.asyncio
+@pytest.fixture
+def single_pool_redis_client():
+    r = get_redis_instance(max_connections=1)
+    yield r
+
+
+@pytest.mark.asyncio
+@pytest.fixture(autouse=True)
+async def traced_redis(redis_client):
+    await redis_client.flushall()
+
+    patch()
+    try:
+        yield
+    finally:
+        unpatch()
+    await redis_client.flushall()
+
+
+def test_patching():
+    """
+    When patching redis library
+        We wrap the correct methods
+    When unpatching  redis library
+        We unwrap the correct methods
+    """
+    assert isinstance(redis.asyncio.client.Redis.execute_command, ObjectProxy)
+    assert isinstance(redis.asyncio.client.Redis.pipeline, ObjectProxy)
+    assert isinstance(redis.asyncio.client.Pipeline.pipeline, ObjectProxy)
+    unpatch()
+    assert not isinstance(redis.asyncio.client.Redis.execute_command, ObjectProxy)
+    assert not isinstance(redis.asyncio.client.Redis.pipeline, ObjectProxy)
+    assert not isinstance(redis.asyncio.client.Pipeline.pipeline, ObjectProxy)
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_basic_request(redis_client):
+    val = await redis_client.get("cheese")
+    assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_unicode_request(redis_client):
+    val = await redis_client.get("😐")
+    assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_decoding_non_utf8_args(redis_client):
+    await redis_client.set(b"\x80foo", b"\x80abc")
+    val = await redis_client.get(b"\x80foo")
+    assert val == b"\x80abc"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_decoding_non_utf8_pipeline_args(redis_client):
+    p = redis_client.pipeline()
+    p.set(b"\x80blah", "boo")
+    p.set("foo", b"\x80abc")
+    p.get(b"\x80blah")
+    p.get("foo")
+
+    response_list = await p.execute()
+    assert response_list[0] is True  # response from redis.set is OK if successfully pushed
+    assert response_list[1] is True
+    assert response_list[2].decode() == "boo"
+    assert response_list[3] == b"\x80abc"
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_long_command(redis_client):
+    length = 1000
+    val_list = await redis_client.mget(*range(length))
+    assert len(val_list) == length
+    for val in val_list:
+        assert val is None
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_override_service_name(redis_client):
+    with override_config("aioredis", dict(service_name="myaioredis")):
```

Comment:
🤔 this test should be failing/not applicable

Reply:
fixed